### ISSUE TITLE
Implement cropping 2D layer

### ISSUE THREAD

**Issue by bergwerf** [deeplearning4j/deeplearning4j#4850]
#### Issue Description

I want to use a cropping layer to remove rows/columns from an input nd array. In my case this only  concerns 2D cropping. I expect this layer to have similar functionality as Cropping2D in Keras, meaning that for a 4d tensor of (batch, width, height, channels) I can crop the width and the height at the beginning and the end by any number of elements.

#### Version Information

Does not seem to be present in the current latest source code.

#### Contributing

N/A

**Comment by maxpumperla** [comment]
I'm totally in on this, as we need it for keras import as well. will tackle it after the next release.

**Comment by AlexDBlack** [comment]
The DL4J layer should be trivial to implement... I was going to try to squeeze this one in before release (i.e., today) - even if we (well, you @maxpumperla ) don't have time to implement keras import for it.

**Comment by maxpumperla** [comment]
@bergwerf cropping2D support is in master now, tomorrow in snaphots

**Comment by bergwerf** [comment]
Very nice! Is there a separate issue I can subscribe to to know when Keras
imports support cropping?

At least for now I'm importing models from Keras because of portability,
implementation speed, and because writing custom loss functions is so neat
in Keras (I hope importing Keras models with custom loss functions already
works?, is there a place where I can find all Keras features supported in
importing?)

**Comment by maxpumperla** [comment]
Let me rephrase: Importing Keras Cropping 2D is now on master (see #4865).

Custom losses are not supported, that's very difficult to do right.

Supported features are here: https://deeplearning4j.org/keras-supported-features (need to add cropping now)

**Comment by bergwerf** [comment]
Ok, thanks!
In that case I will try to avoid custom loss functions. I was mainly using weighted cross entropy, perhaps this is implemented as standard case? I will investigate the DL4J loss functions features a bit further.

**Comment by maxpumperla** [comment]
@bergwerf we have this "one level down" https://github.com/deeplearning4j/nd4j/blob/cc33606b2c0bc297952fda79d0ff1962cde78c46/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/accum/WeightedCrossEntropyLoss.java

If you feel up to adding it to DL4J losses, you can give it a shot.
